core/src/edu/rice/comp413fall2020grey/Replication/Messages/UpdateMessage.java

```diff
@@ -0,0 +1,49 @@
+package edu.rice.comp413fall2020grey.Replication.Messages;
+
+import java.util.Date;
+import java.util.Set;
+import java.util.UUID;
+
+/**
+ * Message for object updates. May be sent from either a replica or
+ * a primary object. The target object is the target for the changes.
+ */
+public class UpdateMessage extends ReplicaMessage {
+
+  /**
+   * Authorizing object for update.
+   */
+  private final UUID authorObject;
+
+  /**
+   * A set of strings representing object updates.
+   */
+  private final Set<String> changes;
```

Comment:
How are these string changes going to be parsed by the replica it seems a little confusing to me.

Reply:
This is a good question, and is probably worth talking about between Replication and Object Storage. It may also affect Object Location since these changes will need to propagate to the registrar. I will add it to the agenda for Monday.